File: app/src/main/java/com/example/capstonedesign_geo/ui/adapter/ChatAdapter.java
package com.example.capstonedesign_geo.ui.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.capstonedesign_geo.R;
import com.example.capstonedesign_geo.data.model.ChatMessage;

import java.util.List;

public class ChatAdapter extends RecyclerView.Adapter<ChatAdapter.MyViewHolder> {
    List<ChatMessage> chatMessageList;
    public ChatAdapter(List<ChatMessage> chatMessageList) {
        this.chatMessageList = chatMessageList;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_chat_message, null);
        MyViewHolder myViewHolder = new MyViewHolder(view);
        return myViewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        ChatMessage chatMessage = chatMessageList.get(position);
        if (chatMessage.getSender().equals("me")) {
            holder.geoImg.setVisibility(View.GONE);
            holder.geoName.setVisibility(View.GONE);
            holder.botMessage.setVisibility(View.GONE);
            holder.userMessage.setVisibility(View.VISIBLE);
            holder.userMessage.setText(chatMessage.getMessage());
        } else {
            holder.geoImg.setVisibility(View.VISIBLE);
            holder.geoName.setVisibility(View.VISIBLE);
            holder.userMessage.setVisibility(View.GONE);
            holder.botMessage.setVisibility(View.VISIBLE);
            holder.botMessage.setText(chatMessage.getMessage());
        }
    }

    @Override
    public int getItemCount() {
        return chatMessageList.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        ImageView geoImg;
        TextView geoName;
        TextView botMessage;
        TextView userMessage;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            geoImg = itemView.findViewById(R.id.geoImg);
            geoName = itemView.findViewById(R.id.geoName);
            botMessage = itemView.findViewById(R.id.botMessage);
            userMessage = itemView.findViewById(R.id.userMessage);
        }
    }
}

File: app/src/main/java/com/example/capstonedesign_geo/ui/adapter/ChatAdapter.kt
package com.example.capstonedesign_geo.ui.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.capstonedesign_geo.data.chat.ChatMessage
import com.example.capstonedesign_geo.databinding.ItemChatMessageBinding

// ChatAdapter: 채팅 메시지를 표시하는 리사이클러뷰 어댑터
class ChatAdapter(private val messages: List<ChatMessage>) : RecyclerView.Adapter<ChatAdapter.ChatViewHolder>() {

    // ChatViewHolder: 각 메시지 뷰를 보관하는 뷰홀더 클래스
    class ChatViewHolder(val binding: ItemChatMessageBinding): RecyclerView.ViewHolder(binding.root)

    // onCreateViewHolder: 뷰홀더를 생성하는 메서드
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ChatViewHolder {
        // ItemChatMessageBinding을 사용하여 뷰바인딩을 설정
        val binding = ItemChatMessageBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        // 생성된 뷰홀더를 반환
        return ChatViewHolder(binding)
    }

    // onBindViewHolder: 뷰홀더에 데이터를 바인딩하는 메서드
    override fun onBindViewHolder(holder: ChatViewHolder, position: Int) {
        val message = messages[position]        // 현재 메시지를 가져옴
        holder.binding.apply {                  // 뷰바인딩을 사용하여 각 뷰에 데이터를 설정
            messageText.text = message.text     // 메시지 텍스트 설정
        }
    }

    override fun getItemCount(): Int = messages.size // 메시지 목록의 크기를 반환
}